packages/hub/src/lib/file-download-info.ts

```diff
@@ -84,9 +92,24 @@ export async function fileDownloadInfo(
 		throw new InvalidApiResponseFormatError("Invalid file size received");
 	}
 
+	let xetInfo: XetInfo | undefined;
+	if (resp.headers.get("Content-Type") === "application/vnd.xet-fileinfo+json") {
+		const json: { casUrl: string; hash: string; refreshUrl: string } = await resp.json();
+
+		xetInfo = {
+			hash: json.hash,
+			refreshUrl: json.refreshUrl,
+		};
+	}
+
 	return {
 		etag,
 		size,
-		downloadLink: new URL(resp.url).hostname !== new URL(hubUrl).hostname ? resp.url : null,
+		xet: xetInfo,
+		// Cannot use resp.url in case it's a S3 url and the user adds an Authorization header to it.
+		url:
+			new URL(resp.url).hostname === new URL(hubUrl).hostname || resp.headers.get("X-Cache")?.endsWith(" cloudfront")
```

Comment:
@XciD @rtrompier can you add `X-Cache` and `Etag` to `Access-Control-Expose-Headers` on our cloudfronts?

```
Access-Control-Allow-Origin: *
Access-Control-Expose-Headers: X-Cache, ETag
```

cc @Michellehbn for viz